### ISSUE TITLE
Utop hangs, ocamlrun works, just 'ocaml' works - waiting on a mutex

### ISSUE THREAD

**Issue by Charlesetc** [diml/utop#189]
This is what utop is printing:
```
$ utop
---------------+--------------------------------------------------------------+---------------
               | Welcome to utop version 1.19.3 (using OCaml version 4.02.3)! |               
               +--------------------------------------------------------------+               
```

Now I tried:

```
$ sudo gdb -p 8635
Attaching to process 8635
[New LWP 8636]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
0x00007efc9ad27903 in select () from /usr/lib/libc.so.6
(gdb) list
1       src/lTerm_term_stubs.c: No such file or directory.
(gdb) where
#0  0x00007efc9ad27903 in select () from /usr/lib/libc.so.6
#1  0x00007efc9a208d7e in unix_select ()
   from /home/charles/.opam/4.02.3/lib/ocaml/stublibs/dllunix.so
#2  0x000000000041c20b in caml_interprete ()
#3  0x000000000041dac8 in caml_main ()
#4  0x000000000041ac1b in main ()
```

Here's a gist with the output of strace:
https://gist.github.com/Charlesetc/9607f5c1df310e1664600f5dd6233cb4

All the 'futex' calls make me think it's waiting on a mutex.

Have you seen this before / know what's going on? I've tried to uninstall and reinstall utop and it's still a problem.

**Comment by Charlesetc** [comment]
Also, my ocamlinit file is empty.

**Comment by Charlesetc** [comment]
So I built utop master and was having the same problem. Then with ocamldebug, stepped through until it starting hanging.

Here is one file:
https://github.com/ocsigen/lwt/blob/master/src/unix/lwt_main.ml
I got to this line:
```
        Lwt_engine.iter (Lwt.paused_count () = 0 && Lwt_sequence.is_empty yielded);
```

This calls engine.iter which is defined in this file: https://github.com/ocsigen/lwt/blob/6c7a95a83cc855c472d0a05530d023285d49f8a7/src/unix/lwt_engine.ml

The `engine` here is a type of this class:
```
class select = object
  inherit select_based

  method private select fds_r fds_w timeout =
    let fds_r, fds_w, _ = Unix.select fds_r fds_w [] timeout in
    (fds_r, fds_w)
end
```

And then eventually it calls the 'select' method. The Unix.select is what's hanging...

Is this an Lwt bug instead? I'm still not really sure what's happening here.
